### PR TITLE
22667 - EFT interm statements payment method incorrect

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:*
https://github.com/bcgov/entity/issues/22667

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/tests/unit/services/test_statement.py [hunk 1]
```diff
@@ -701,104 +698,3 @@ def localize_date(date: datetime):
     """Localize date object by adding timezone information."""
     pst = pytz.timezone('America/Vancouver')
     return pst.localize(date)
-
-
-def test_statement_various_payment_methods_history(db, app):
```
Comment (seeker25) [inline pay-api/tests/unit/services/test_statement.py:706]: not needed wont work anyways and cause future pains

File: jobs/payment-jobs/tasks/statement_task.py [hunk 2]
```diff
@@ -156,7 +157,8 @@ def _create_statement_records(cls, search_filter, statement_settings, account_ov
             to_date=statement_to,
             notification_status_code=NotificationStatus.PENDING.value
             if pay_account.statement_notification_enabled is True and cls.has_date_override is False
-            else NotificationStatus.SKIP.value
+            else NotificationStatus.SKIP.value,
+            payment_methods=StatementService.get_payment_methods_from_details(invoice_detail_tuple, pay_account)
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/statement_task.py:161]: We can gun this, if we consolidate the statements into one document

File: pay-api/migrations/versions/2024_08_08_5cb9c5f5896c_.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+# Please ignore those lines and don't include in migration.
+
+revision = '5cb9c5f5896c'
+down_revision = '17ca5cd561ca'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    with op.batch_alter_table('statements', schema=None) as batch_op:
```
Comment (seeker25) [inline pay-api/migrations/versions/2024_08_08_5cb9c5f5896c_.py:25]: We can gun this, if we consolidate the statements into one document

File: pay-api/src/pay_api/services/statement.py [hunk 4]
```diff
@@ -471,6 +399,15 @@ def populate_overdue_from_invoices(statements: List[StatementModel]):
             statement.is_overdue = overdue_statements.get(statement.id, 0) > 0
         return statements
 
+    @staticmethod
+    def get_payment_methods_from_details(invoice_detail_tuple, pay_account) -> str:
```
Comment (seeker25) [inline pay-api/src/pay_api/services/statement.py:403]: We can gun this, if we consolidate the statements into one document

**Comment by seeker25** [comment]
TODO, need some unit tests for `statement.payment_methods` before merging

File: pay-api/src/pay_api/services/statement.py [hunk 5]
```diff
@@ -128,68 +127,6 @@ def asdict(self):
         d = statement_schema.dump(self._dao)
         return d
 
-    @staticmethod
-    def get_payment_methods_query(statement: StatementModel):
```
Comment (seeker25) [inline pay-api/src/pay_api/services/statement.py:132]: Wont work for payments that don't show up in the `payment_account_history` table

File: pay-api/src/pay_api/services/statement.py [hunk 6]
```diff
@@ -471,6 +399,15 @@ def populate_overdue_from_invoices(statements: List[StatementModel]):
             statement.is_overdue = overdue_statements.get(statement.id, 0) > 0
         return statements
 
+    @staticmethod
+    def get_payment_methods_from_details(invoice_detail_tuple, pay_account) -> str:
+        """Grab payment methods from invoices detail tuple."""
+        payment_methods = {payment_method for _, payment_method, _,
+                           auth_account_id in invoice_detail_tuple if pay_account.auth_account_id == auth_account_id}
+        if not payment_methods:
+            payment_methods = {pay_account.payment_method}
```
Comment (ochiu) [inline pay-api/src/pay_api/services/statement.py:408]: Will this work for old empty statements that is not the current payment method? 

For example, if I am currently EFT and I view an empty statement from a few months ago that is another payment method.

Comment (seeker25) [reply pay-api/src/pay_api/services/statement.py:408]: Yes that's exactly the case this covers

Comment (seeker25) [reply pay-api/src/pay_api/services/statement.py:410]: Should be fixed take a look again

**Review by ochiu — APPROVED** [review]
Awesome thanks for adding the re-use statement logic when regenerating a statement

**Pull request merged**